low-level/sensors/impl/centos_7/disk_monitor.py

```diff
@@ -88,7 +88,7 @@ class DiskMonitor(SensorThread, InternalMsgQ):
     DISK_FAULT_RESOLVED_ALERT_TYPE = "fault_resolved"
     DRIVE_DBUS_INFO = 'dbus_info'
     DRIVE_FAULT_ATTR = 'smart_attributes'
-    NODE_DISK_RESOURCE_TYPE = "server:hw:disk"
+    NODE_DISK_RESOURCE_TYPE = "server:hw:disk:smart"

```

Comment:
For insertion/removal alerts we are using dbus api's, smartctl we are using only for fault detection.
so this might be misguiding for insertion/removal alerts I think.

Reply:
Yes, right.
Then `Why smart?`, disk monitor sensor does process_msg() only for
 sensor_request_type == "simulate_failure"    --> SMART
 sensor_request_type == "resend_drive_status" --> Perhaps systemd
 sensor_request_type == "disk_smart_test"   --> SMART

systemd manager tells a drive is local and physical, also its path to find out if that drive is existing one or added newly.

Reply:
>For insertion/removal alerts we are using dbus api's, smartctl we are using only for fault detection.
so this might be misguiding for insertion/removal alerts I think.

Good point indeed.

Problem is same disk resource is monitored by two different sensors (IPMI based nodehw sensor, active on dell, hpe, intel servers) and this disk monitor and separate alerts raised for same. At some point we should club these alerts into one and put these disk details coming from either IPMI / smartctl/ systemd as specific info, so whatever data is available can be supplied, like on supermicro servers, disk sensor doesnt exists, so IPMI sensor cant detect any disk present or fault.

Until then, separate sensor need to use separate resource_type. 
for disk fault -> "server:hw:disk:smart"
for disk presence -> "server:hw:disk:device" !? ( this would complicate stuff for us, but can be better experience for user)

>sensor_request_type == "simulate_failure" --> SMART
sensor_request_type == "resend_drive_status" --> Perhaps systemd
sensor_request_type == "disk_smart_test" --> SMART

These are all for legacy logic

>systemd manager tells a drive is local and physical,

This logic also needs improvement, interesting study to do, based on SCSI Ids i still think we can figure out

> also its path to find out if that drive is existing one or added newly.

Am not sure how, based on path ?

Reply:
~~systemd manager~~ Disk manager(using system bus interfaces) provides the drive path. This, we store in cache and call the disk manager after some interval. Then compare the latest drive information to report whats removed and added.
```
{
                    "/org/freedesktop/UDisks2/drives/drive_3": {
                        "dbus_info": dbus_properties,
                        "node_disk": False
                    },
                    "/org/freedesktop/UDisks2/drives/ST1000NM0055_1V410C_ZBS1VJHX": {
                        "dbus_info": dbus_properties,
                        "node_disk": True
                    }
                }
```

Reply:
Yes exactly, using Dbus api's are adding `insertion/removal` event signals under the path `/org/freedesktop/UDisks2/drives/`, so that the alerts are raised for these events. Alert will be published with `disk:smart` but it is not actually detected by the `smartctl` that is the case I wanted to put. But if it's only to distinguish the sensors, then it's fine I guess. or we can replace `smart` with `dbus` or something similar for insertion & removal alerts.

Reply:
>Then compare the latest drive information to report whats removed and added.

Ok, got it.

>But if it's only to distinguish the sensors, then it's fine I guess.

Alert not supposed to tell backend sensor. 

>"server:hw:disk:smart"

smart here is to tell fault detected through SMART tests, which is industry wide known approach

resource type needs to tell about resource, and not the backend. so dbus is not needed, but something related to resource itself, so had suggested this
>for disk presence -> "server:hw:disk:device"

Reply:
`server:hw:disk:device` 
This can for for disk_monitor sensor, instead of exposing the technique in resource type.

Reply:
yes..

Reply:
Done.

Reply:
Am not clear, as i see in code, "server:hw:disk:device" is used as resource_type for alerts (fault or presence) and we are not using "server:hw:disk:smart" any more !?

Reply:
Yes, @cdeshmukh.

Reply:
ok